client/packages/mobile/app/src/main/java/org/openmsupply/mobile/MainActivity.java

```diff
@@ -0,0 +1,294 @@
+package org.openmsupply.mobile;
+
+import static android.webkit.WebSettings.MIXED_CONTENT_ALWAYS_ALLOW;
+
+import android.content.Context;
+import android.net.http.SslError;
+import android.net.nsd.NsdManager;
+import android.net.nsd.NsdServiceInfo;
+import android.os.Bundle;
+import android.print.PrintAttributes;
+import android.print.PrintDocumentAdapter;
+import android.print.PrintManager;
+import android.provider.Settings;
+import android.util.Log;
+import android.webkit.ConsoleMessage;
+import android.webkit.JavascriptInterface;
+import android.webkit.SslErrorHandler;
+import android.webkit.WebChromeClient;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openmsupply.mobile.RemoteServer;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Random;
+
+public class MainActivity extends AppCompatActivity implements NsdManager.DiscoveryListener {
+    static final String TAG_WV = "WEB-VIEW";
+    static final String SERVICE_TYPE = "_omsupply._tcp";
+    static final String SERVICE_NAME = "omSupplyServer";
+    static final String PROTOCOL_KEY = "protocol";
+    static final String CLIENT_VERSION_KEY = "client_version";
+    static final String HARDWARE_ID_KEY = "hardware_id";
+    static final Integer PORT = 8000;
+
+    RemoteServer server = new RemoteServer();
+    NsdManager discoveryManager;
+    WebView wv;
+    WebView printView;
+    JSONArray discoveredServers;
```

Comment:
Thought to use JSON array and object for servers, could make a class for them, but this was quicker for me, haven't done java in a bit